internal/graph/graph_test.go

```diff
@@ -111,6 +111,22 @@ func TestGraphString(t *testing.T) {
 	require.Contains(t, g.String(), "*graph.Parent12 -> (object) obj: *graph.Parent12")
 }
 
+func TestCtorConflicts(t *testing.T) {
+	t.Parallel()
+	g := NewGraph()
+
+	err := g.InsertConstructor(func() (*Child1, *Child1, *Child3, error) {
+		return &Child1{}, &Child1{}, &Child3{}, nil
+	})
+	require.NoError(t, err)
```

Comment:
what do you mean? First InsertConstructor will always succeed since return objects are not present in constructor, followed by the error (line 123 t0 127)